src/bsoncxx/json.cpp

```diff
@@ -274,15 +274,16 @@ std::string to_json(types::value value) {
 
     json_visitor v(ss, false, 0);
 
-    switch ((int)value.type()) {
-#define BSONCXX_ENUM(name, val)            \
-    case val:                              \
-        v.visit_value(value.get_##name()); \
-        break;
+    if(value.type()) {
```

Comment:
ah yeah, I forgot they had the overload for a standalone value.

 However, consider the case in which a view is passed in to `to_json`. The view should not have been able to have been created with an 'invalid' value as we check for that when appending to a builder.

So maybe we throw here, but in the implementation of `to_json(document::view view)`

```
try {
.. existing logic
} catch (/* exception thrown for invalid value */) {
  BSONCXX_UNREACHABLE;
}
```